### ISSUE TITLE
Should we throw an early error for function calls on LHS in strict mode?

### ISSUE THREAD

**Issue by zenparsing** [Microsoft/ChakraCore#6283]
SpiderMonkey throws an early error in the following case:

```js
function f() {}
function g() { 'use strict'; if (false) f() = 1;  }
```

whereas CC and V8 do not. Should we throw early as well?

**Comment by fatcerberus** [comment]
What does the specification say?  I can’t check right this minute but I’ll look later tonight.

**Comment by pleath** [comment]
The spec says throw. (See 12.3.1.6.) foo()[x] = 0 and foo().x = 0 are okay, but not foo() = 0. The trouble is, there's a lot of code on the web that contains statements like foo(i) = n. Chakra will always throw if it executes such a statement, but we can't get away with raising an early error. V8 and SpiderMonkey seem to have run into the same compat issue -- in sloppy mode, anyway.

**Comment by fatcerberus** [comment]
> ...but we can't get away with raising an early error.

Yeah, I wasn't totally clear above: I knew this was an error at runtime, but I thought I remembered that it *wasn't* an *early error* because (at least at one time) the specification allowed host-defined functions to return references (think C++-style references) which could be used as an assignment target.  I believe that was forbidden in ES2015, but obviously legacy code remains that takes advantage of the syntax being legal.

**Comment by pleath** [comment]
Right. I don't know when it was officially made bad syntax, but the horse left the barn long before.
